pilot/pkg/serviceregistry/kube/deregister.go

```diff
@@ -78,7 +79,8 @@ func DeRegisterEndpoint(client kubernetes.Interface, namespace string, svcName s
 			If the service endpoint has not been registered
 			before, report proper error message.
 		*/
-		log.Errora("Could not find ip %s in svc %s endpoints", ip, svcName)
+		err = fmt.Errorf("could not find ip %s in svc %s endpoints", ip, svcName)
+		log.Errora(err)
 		return err
```

Comment:
Good find, this was actually a bug before - as it would always return `nil` in this case